src/externs.ts

```diff
@@ -79,26 +86,114 @@ export function getGeneratedExterns(externs: {[fileName: string]: string}): stri
   return allExterns;
 }
 
+/**
+ * Returns a mangled version of the module name (resolved file name) for source file.
+ *
+ * The mangled name is safe to use as a JavaScript identifier. It is used as a globally unique
+ * prefix to scope symbols in externs file (see code below).
+ *
+ * @param contextFileName if given is used as the context file path to resolve the source file's
+ *     name against.
+ */
+export function moduleNameAsIdentifier(
+    host: AnnotatorHost, fileName: string, contextFileName?: string): string {
+  const resolved = resolveModuleName(host, contextFileName || '', fileName);
+  return host.pathToModuleName('', resolved).replace(/\./g, '$');
+}
+
+/**
+ * isInGlobalAugmentation returns true if declaration is the immediate child of a 'declare global'
```

Comment:
Good point. We don't currently detect that (and it's a bit tricky - would we walk up all declarations and find whether any parent declaration is global). The code falls into the `else` case and reports and error though, because `a.b` is a property access expression. I think that's good enough for now.